orderbook/src/fee.rs

```diff
@@ -44,6 +62,31 @@ impl MinFeeCalculator {
     // Returns an error if there is some estimation error and Ok(None) if no information about the given
     // token exists
     pub async fn min_fee(&self, token: H160) -> Result<Option<Measurement>> {
+        let now = (self.now)();
+        let official_valid_until = now + Duration::seconds(STANDARD_VALIDITY_FOR_FEE_IN_SEC);
+        let internal_valid_until = now + Duration::seconds(PERSISTED_VALIDITY_FOR_FEE_IN_SEC);
+
+        if let Ok(Some(past_fee)) = self
+            .measurements
+            .get_min_fee(token, official_valid_until)
+            .await
+        {
+            return Ok(Some((past_fee, official_valid_until)));
+        }
+
+        let min_fee = match self.compute_min_fee(token).await? {
+            Some(fee) => fee,
+            None => return Ok(None),
+        };
+
+        let _ = self
+            .measurements
+            .save_fee_measurement(token, internal_valid_until, min_fee)
+            .await;
+        Ok(Some((min_fee, official_valid_until)))
```

Comment:
So I guess the meat of this logical change is right here, where we check the DB for a non-expired min fee and use that if available. Do we expect that the the validity period will be something we inevitably have to maintain as the ecosystem changes?

Reply:
Yes, the API will have to guarantee that it accepts a fee that it originally quoted for some time (to allow for the time between quoting and signing the order without rejecting the order if the gas price went up in the meantime). The concrete duration of this time may change.